packages/embed/src/utils/create-iframe/create-iframe.ts

```diff
@@ -18,6 +19,13 @@ export const createIframe = (formId: string, type: EmbedType, options: CreateIfr
   window.addEventListener('message', getFormQuestionChangedHandler(embedId, options))
   window.addEventListener('message', getFormSubmitHandler(embedId, options))
 
+  if (options.shareGoogleAnalyticsInstance) {
+    window.addEventListener('message', (ev) => {
+      if (ev.data.type === 'form-ready') {
+        setupGoogleAnalyticsInstanceSharingFeature(iframe, embedId)
+      }
+    })
+  }
```

Comment:
I am not sure this is the best approach, but previously we would call it when the form is ready

Reply:
Please reuse `getFormReadyHandler` handle (refactor it to take callback as 2nd param instead of "options" object) for this. This will validate the embed id and listen to events from correct form. This is necessary when there are multiple embeds on single page.